snapcraft/storeapi/__init__.py

```diff
@@ -280,15 +335,20 @@ def __init__(self, conf):
             'UBUNTU_STORE_API_ROOT_URL',
             constants.UBUNTU_STORE_API_ROOT_URL))
 
+    def get_macaroon(self, acls):
+        response = self.post(
+            'acl/',
+            data=json.dumps({'permissions': acls}),
+            headers={'Content-Type': 'application/json'})
+        if response.ok:
+            return response.json()['macaroon'], None
+        else:
+            return None, response.text
+
     def snap_upload(self, data):
-        session = common.get_oauth_session(config.Config())
-        if session is None:
-            raise InvalidCredentialsError()
-        url = urllib.parse.urljoin(
-            self.root_url, 'click-package-upload/{}/'.format(
-                urllib.parse.quote_plus(data['name'])))
-        return session.post(url, data)
-
-    def get(self, download_url):
-        session = common.get_oauth_session(config.Config())
-        return session.get(download_url)
+        auth = _macaroon_auth(self.conf)
+        response = self.post(
+            'snap-upload/', data=json.dumps(data),
```

Comment:
The endpoint is now called snap-push ([bug report](https://bugs.launchpad.net/snapcraft/+bug/1587165)); there seems little point in adding new code that refers to the old endpoint.